routes/payload.js

```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+
+const router = express.Router();
+
+router.post('/', (req, res) => { 
+    const input = req.body;
+    var strPayload = JSON.stringify(input.payload);
+    const keys = Object.keys(input.referenceData);
+    for(var i=0; i<keys.length; i++){
+        var reg = new RegExp(`{${keys[i]}}`, "g");
+       strPayload = strPayload.replace(reg, input.referenceData[keys[i]]);
```

Comment:
this works for smaller sized request data.
can you check with a larger input file(say 4MB json) and let me know if your code runs without slowing down the system?
Also if you find it slow, use json object iteration/ walkthrough method to replace placeholders and compare the system load?